### PR TITLE
Fixed thumbnails

### PR DESCRIPTION
**Pull request by phenaproxima**
This fixes https://www.drupal.org/node/2619906. I used Google's API Explorer to generate the request URL, so that's as good as documented :)

This PR is very light on error handling, because I'm not sure what the best course of action is in the event of an HTTP error (or some other problem with the response that comes back from YouTube). I figure that error handling can be beefed up in the future, though...

### CODE REVIEW

File: config/schema/media_entity_embeddable_video.schema.yml [hunk 1]
```diff
@@ -17,6 +17,13 @@ media_entity_embeddable_video.settings:
     wochit_progn:
       type: string
       label: 'Publisher information used by Wochit ("progn" argument in embed code)'
+    youtube:
+      type: mapping
+      label: 'YouTube'
+      mapping:
+        api_key:
+          type: string
```
Comment (webflo) [inline config/schema/media_entity_embeddable_video.schema.yml:25]: I think the data-type in entity schema and default value in media_entity_embeddable_video.settings does not match.

Comment (phenaproxima) [inline config/schema/media_entity_embeddable_video.schema.yml:25]: Good point. Fixed.

File: src/Plugin/MediaEntity/VideoProvider/YouTube.php [hunk 2]
```diff
@@ -44,24 +86,14 @@ public function defaultConfiguration() {
    * {@inheritdoc}
    */
   public function thumbnailURI() {
-    $maxres_thumb = 'http://img.youtube.com/vi/' . $this->matches['id'] . '/maxresdefault.jpg';
```
Comment (jcisio) [inline src/Plugin/MediaEntity/VideoProvider/YouTube.php:47]: FYI this is how I get the thumbnail URL in a library that I maintain https://github.com/jcisio/MediaWrapper/blob/master/MediaWrapper/Wrapper/Youtube.php#L55 (maxresdefault, then mqdefault). Again, I don't have source documenting it...

**Comment by slashrsm** [comment]
Squshed commits, reworded commit message and pushed. 

https://github.com/drupal-media/media_entity_embeddable_video/commit/3a553c805ec21187377e49f32757be12e1398fcf

**Pull request closed without merging**